src/runtime/cuda/cuda_device_api.cc

```diff
@@ -30,7 +30,10 @@ class CUDADeviceAPI final : public DeviceAPI {
                 &value, cudaDevAttrMaxThreadsPerBlock, ctx.device_id)
             == cudaSuccess);
         break;
-      case kMaxThreadsPerBlock: {
+    case kStreamAPI:
```

Comment:
I feel maybe we don't need stream API query. Do a 
```
try:
    TVMStreamCreate()
except:
   pass
```
will be sufficient to get the info without hacking into DeviceAPI too deep

Reply:
Because we already have exception handling, this can be done from front end

Reply:
It is not required, no.  It depends if the error is ambiguous.  There is a difference between not supporting streams at all and failing to create one for a specific potentially user-fixable reason (like you have created too many already).  Furthermore when debugging something being able to break on exceptions and having them actually mean there is an error condition is helpful.

Either way, at device initialization the same query step can be done a couple different ways so  I can remove it.  Clients will have to have a 'stream unsupported' pathway either way.